scripts/linking.py

```diff
@@ -289,37 +297,147 @@ def get_mention_text_and_ids(data: List[data_util.MedMentionExample],
+                doc = nlp(example.text)
+                ner_mentions = [ent.text for ent in doc.ents]
+                doc_candidates = set()
+                doc_golds = set(gold_umls_ids)
+
+                # it is possible that a spacy model does not find any entities in an abstract
+                if ner_mentions == []:
+                    entity_missed_count += len(doc_golds)
+                    all_golds_per_doc_set += list(doc_golds)
+                    all_golds += gold_umls_ids
+                    continue
+
+                batch_candidate_neighbor_ids = candidate_generator.generate_candidates(ner_mentions, k)
+
+                for ner_mention, candidate_neighbor_ids in zip(ner_mentions, batch_candidate_neighbor_ids):
+                    # Keep only canonical entities for which at least one mention has a score less than the threshold.
+                    filtered_ids = {k: v for k, v in candidate_neighbor_ids.items() if any([z[1] <= threshold for z in v])}
+                    num_candidates.append(len(candidate_neighbor_ids))
+                    num_filtered_candidates.append(len(filtered_ids))
+
+                    doc_candidates.update(filtered_ids)
+
+                    if len(filtered_ids) == 0:
+                        mention_no_links_count += 1
+
+                # the number of correct entities for a given document is the number of gold entities contained in the candidates
+                # produced for that document
+                entity_correct_links_count += len(doc_candidates.intersection(doc_golds))
+                # the number of incorrect entities for a given document is the number of gold entities not contained in the candidates
+                # produced for that document
+                entity_missed_count += len(doc_golds - doc_candidates)
+                
+                all_golds_per_doc_set += list(doc_golds)
+                all_golds += gold_umls_ids
+                all_mentions += ner_mentions
+
+            print(f'MedMentions entities not in UMLS: {len(missing_entity_ids)}')
+            print(f'MedMentions entities found in UMLS: {len(all_golds)}')
+            print(f'K: {k}, Filtered threshold : {threshold}')
+            print('Gold concept in candidates: {0:.2f}%'.format(100 * entity_correct_links_count / len(all_golds_per_doc_set)))
+            print('Gold concepts missed: {0:.2f}%'.format(100 * entity_missed_count / len(all_golds_per_doc_set)))
+            print('Candidate generation failed: {0:.2f}%'.format(100 * mention_no_links_count / len(all_mentions)))
+            print("Mean, std, min, max candidate ids: ", np.mean(num_candidates), np.std(num_candidates), np.min(num_candidates), np.max(num_candidates))
+            print("Mean, std, min, max filtered candidate ids: ", np.mean(num_filtered_candidates), np.std(num_filtered_candidates), np.min(num_filtered_candidates), np.max(num_filtered_candidates))
+
```

Comment:
Can you also print the time it took to run this function. I think it will be a good estimate for the run time of the full pipeline